module/zfs/vdev.c

```diff
@@ -3454,18 +3464,12 @@ vdev_set_state(vdev_t *vd, boolean_t isopen, vdev_state_t state, vdev_aux_t aux)
 	}
 
 	/*
-	 * Notify ZED of any significant state-change on a leaf vdev.
-	 *
-	 * We ignore transitions from a closed state to healthy unless
-	 * the parent was degraded.
+	 * Notify ZED of any state change to a leaf vdev
 	 */
 	if (vd->vdev_ops->vdev_op_leaf &&
-	    ((save_state > VDEV_STATE_CLOSED) ||
-	    (vd->vdev_state < VDEV_STATE_HEALTHY) ||
-	    (vd->vdev_parent != NULL &&
-	    vd->vdev_parent->vdev_prevstate == VDEV_STATE_DEGRADED))) {
+	    ((save_state != VDEV_STATE_UNKNOWN) &&
+	    (save_state != VDEV_STATE_CLOSED)))
```

Comment:
After addtional testing, I found a better method than snooping the parent to catch state changes after a vdev_clear().
```
@@ -3464,12 +3469,19 @@ vdev_set_state(vdev_t *vd, boolean_t isopen, vdev_state_t state, vdev_aux_t aux)
        }

        /*
-        * Notify ZED of any state change to a leaf vdev
+        * Notify ZED of any significant state change to a leaf vdev
         */
-       if (vd->vdev_ops->vdev_op_leaf &&
-           ((save_state != VDEV_STATE_UNKNOWN) &&
-           (save_state != VDEV_STATE_CLOSED)))
-               zfs_post_state_change(spa, vd, save_state);
+       if (vd->vdev_ops->vdev_op_leaf) {
+               /* preserve original state from a vdev_reopen() */
+               if ((vd->vdev_prevstate != VDEV_STATE_UNKNOWN) &&
+                   (vd->vdev_prevstate != vd->vdev_state) &&
+                   (save_state <= VDEV_STATE_CLOSED))
+                       save_state = vd->vdev_prevstate;
+
+               /* filter out state change due to initial vdev_open */
+               if (save_state > VDEV_STATE_CLOSED)
+                       zfs_post_state_change(spa, vd, save_state);
+       }

        if (!isopen && vd->vdev_parent)
                vdev_propagate_state(vd->vdev_parent);
```